pyomo/common/dependencies.py

```diff
@@ -999,10 +999,13 @@ def _finalize_numpy(np, available):
         # registration here (to bypass the deprecation warning) until we
         # finally remove all support for it
         numeric_types._native_boolean_types.add(t)
-    _floats = [np.float_, np.float16, np.float32, np.float64]
+    _floats = [np.float16, np.float32, np.float64]
     # float96 and float128 may or may not be defined in this particular
     # numpy build (it depends on platform and version).
     # Register them only if they are present
+    if hasattr(np, 'float_'):
+        # Prepend to preserve previous functionality
```

Comment:
Note that `float_`/`complex_` were simple aliases to `float64`/`complex128`, so it is not necessary to handle them for numpy 1.x, just dropping them is fully backwards compatible.